packages/api/portfolios/createPortfolio.ts

```diff
@@ -0,0 +1,101 @@
+import validator from "@middy/validator";
+import { ApiGatewayEventParsed } from "../utils/eventHandlingTool";
+import { APIGatewayProxyResult, Context } from "aws-lambda";
+import { ApiSuccessResponse, SuccessStatusCode } from "../utils/response";
+import { CORS_CONFIGURATION } from "../utils/corsConfig";
+import { createConnection } from "../utils/database";
+import { errorHandlingMiddleware } from "../utils/errorHandlingMiddleware";
+import { IpCheckerMiddleware } from "../utils/ipLogging";
+import { IPortfolio, IPortfolioCreate } from "../../orm/entity/Portfolio";
+import { PortfolioRepository } from "../repository/PortfolioRepository";
+import { validateRequestShape } from "../utils/shapeValidator";
+import { wrapSchema } from "../utils/schemaWrapper";
+import schema = require("../models/internalSchema.json");
+import xssSanitizer from "../utils/xssSanitizer";
+import { EmptyPortfolioMiddleware } from "../utils/emptyPortfolioMiddleware";
+
+/**
+ * Creates a new Portfolio
+ * A new portfolioId will be generated by the database.
+ *
+ * @param event - The POST request from API Gateway
+ */
+export async function baseHandler(
+  event: ApiGatewayEventParsed<IPortfolio>,
+  context?: Context
+): Promise<APIGatewayProxyResult> {
+  const setupResult = validateRequestShape<IPortfolio>(event);
+  const portfolioBody = setupResult.bodyObject;
+
+  // set up database connection
+  const connection = await createConnection();
+  const portfolioRepository = connection.getCustomRepository(PortfolioRepository);
+  let insertedPortfolio: IPortfolioCreate;
+
+  try {
+    const insertResult = await portfolioRepository.createPortfolio(portfolioBody);
+    const portfolioId = insertResult.identifiers[0].id;
+    // query for the newly inserted portfolio
+    insertedPortfolio = await portfolioRepository.findOneOrFail({
+      select: [
+        "id",
+        "name",
+        "csp",
+        "description",
+        "dodComponents",
+        "owner",
+        "portfolioManagers",
+        "createdAt",
+        "updatedAt",
+        "archivedAt",
+        "administrators",
+      ],
+      where: { id: portfolioId },
+    });
+
+    console.log("Inserted Portfolio: " + JSON.stringify(insertedPortfolio));
+  } finally {
+    connection.close();
+  }
+
+  return new ApiSuccessResponse<IPortfolioCreate>(insertedPortfolio, SuccessStatusCode.CREATED);
+}
+
+// a workaround for ensuring validation does not give a false failure due to 'allOf'
+// also a band-aid for ensuring BaseObject properties (e.g., id) are not accepted
+// when sent in the request body
+const { type, additionalProperties, required: portfolioRequired } = schema.Portfolio;
+const { required: baseRequired } = schema.PortfolioBase;
+const { portfolioManagers, description, owner /*, applications */ } = schema.Portfolio.properties;
+const { csp, dodComponents, name } = schema.PortfolioBase.properties;
+const { portfolioAdministrators } = schema.PortfolioAccess.properties;
```

Comment:
This allows `portfolioAdministrators` through the validator, but there is no mapping to `administrators` which is returned after creating the portfolio.  The simplest solution appears to be update the spec like below.  Otherwise it appears the `Portfolio` entity (or another class) has to be updated with `portfolioAdministrators`.

```yaml
PortfolioAccess:
      description: >-
        Represents a set of Operators who should be granted access to
        a Portfolio at a specific access level. Uses an enum construct for forward
        compatibility should other access levels be specified.
      properties:
        administrators:
          type: array
          items:
            type: string
            format: email
```

Reply:
Agree this is the simplest solution.  Changed in 88d13a0c85d67b37279c6ebb41b677da58c467cc